libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationNotification.java

```diff
@@ -86,18 +87,29 @@ void unregisterReceiver(Context context) {
   private void initialize(Context context, MapboxNavigation mapboxNavigation) {
     this.mapboxNavigation = mapboxNavigation;
     etaFormat = context.getString(R.string.eta_format);
-    RouteOptions routeOptions = mapboxNavigation.getRoute().routeOptions();
-    distanceFormatter = new DistanceFormatter(context, routeOptions.language(), routeOptions.voiceUnits());
+    initializeDistanceFormatter(context, mapboxNavigation);
     notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
     isTwentyFourHourFormat = DateFormat.is24HourFormat(context);
     createNotificationChannel(context);
     buildNotification(context);
     registerReceiver(context);
   }
 
-  /**
-   * Notification channel setup for devices running Android Oreo or later.
-   */
+  private void initializeDistanceFormatter(Context context, MapboxNavigation mapboxNavigation) {
+    RouteOptions routeOptions = mapboxNavigation.getRoute().routeOptions();
+    String language;
```

Comment:
What about initializing both `language` and `unitType` with the values from `LocaleUtils` and if the route options are not `null` use the values from `routeOptions`? This way we'd be able to remove the `else` branch and make the code more compact.

Reply:
What we need to balance here is going "clean" vs “performant” (note that we’d need to create `LocaleUtils` instances when sometimes won’t be needed).